microsoft/testsuites/core/storage.py

```diff
@@ -277,6 +292,57 @@ def verify_scsi_disk_controller_type(self, node: RemoteNode) -> None:
     def verify_nvme_disk_controller_type(self, node: RemoteNode) -> None:
         self._verify_disk_controller_type(node, DiskControllerType.NVME)
 
+    @TestCaseMetadata(
+        description="""
+        This test verifies the nvme drivers are either built in or present in initramfs.
+        This test can be used to verify an image before it is tagged as NVMe capable.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            supported_platform_type=[AZURE],
+        ),
+    )
+    def verify_supports_nvme(self, node: RemoteNode, environment: Environment) -> None:
+        has_nvme_core = node.tools[KernelConfig].is_built_in("CONFIG_NVME_CORE") or (
+            node.tools[KernelConfig].is_built_as_module("CONFIG_NVME_CORE")
+            and node.tools[Lsinitrd].has_module("nvme-core.ko")
+        )
+        has_nvme = node.tools[KernelConfig].is_built_in("CONFIG_BLK_DEV_NVME") or (
+            node.tools[KernelConfig].is_built_as_module("CONFIG_BLK_DEV_NVME")
+            and node.tools[Lsinitrd].has_module("nvme.ko")
+        )
+
+        # Pass condition
+        if has_nvme_core and has_nvme:
+            return
+
+        assert isinstance(environment.platform, AzurePlatform)
+        node_runbook: AzureNodeSchema = node.capability.get_extended_runbook(
+            AzureNodeSchema, AZURE
+        )
+        image = node_runbook.image
+        # Failure conditions
+        if image and not isinstance(image, VhdSchema):
```

Comment:
VHD should be able to set in the runbook to specify it supports nvme.

Reply:
Can you list the situations in a table? With current logic, the test case is always disabled for vhd.

Reply:
|                             | Marketplace | SIG / CGI | VHD                 |
|-----------------------------|-------------|-----------|---------------------|
| Has drivers                 | PASS        | PASS      | PASS                |
| Missing drivers, not tagged | SKIP        | SKIP      | SKIP                |
| Missing drivers, tagged capable     | FAIL        | FAIL      | VHD is never tagged |

Reply:
The main purpose of this test case is "Tag the image if and only if the test case passes". We don't really care about the skip vs. failure condition unless an image published in the marketplace is tagged but doesn't have drivers. This test case does not really care about VHDs because they aren't tagged.

Reply:
Here is how I see the options
|                            | Deploying VHD on NVMe | Running verify_support_nvme  |
|----------------------------|-----------------------|------------------------------|
| Default support NVMe       |         Works!        | Why is test case failing? :( |
| Default don't support NVMe |   Doesn't deploy :(   |       Catches Failures       |

I could add a tag which it separate from `disk_controller_type` for the image, but it would be quite niche use case. It would be confusing and I don't think it would really be used.

Reply:
If we add a test variable for the test case, can it solve the problem?